src/ros.c

```diff
@@ -346,8 +353,23 @@ int main (int argc, char **argv) /* {{{ */
 	if (passwd == NULL)
 		exit (EXIT_FAILURE);
 
-	c = ros_connect (host, ROUTEROS_API_PORT,
-			opt_username, passwd);
+	if(0 != opt_timeout)
+	{
+		// Timeout is set to something, use it!
+		struct timeval timeout;
```

Comment:
I recommend using a struct initializer here:

```c
struct timeval timeout = {
  .tv_sec = opt_timeout,
};
```

This automatically initializes the rest of the struct to zero.